--- origsrc/qtbase-opensource-src-5.1.0/mkspecs/features/qt_tool.prf	2013-07-02 02:09:32.000000000 -0500
+++ src/qtbase-opensource-src-5.1.0/mkspecs/features/qt_tool.prf	2013-08-14 15:34:36.410627200 -0500
@@ -43,3 +43,5 @@ CONFIG += console
         cache(QT_TOOL.$${MODULE}.$$var, transient)
 
 }
+
+LIBS_PRIVATE += -lz
--- origsrc/qtbase-opensource-src-5.1.0/src/plugins/platforms/minimal/minimal.pro	2013-07-02 02:09:53.000000000 -0500
+++ src/qtbase-opensource-src-5.1.0/src/plugins/platforms/minimal/minimal.pro	2013-08-14 16:22:29.645966800 -0500
@@ -5,6 +5,7 @@ PLUGIN_CLASS_NAME = QMinimalIntegrationP
 load(qt_plugin)
 
 QT += core-private gui-private platformsupport-private
+LIBS_PRIVATE += $$QT_LIBS_GLIB
 
 SOURCES =   main.cpp \
             qminimalintegration.cpp \
--- origsrc/qtbase-opensource-src-5.1.0/src/plugins/platforms/offscreen/offscreen.pro	2013-07-02 02:09:52.000000000 -0500
+++ src/qtbase-opensource-src-5.1.0/src/plugins/platforms/offscreen/offscreen.pro	2013-08-14 16:27:41.790820500 -0500
@@ -19,6 +19,7 @@ OTHER_FILES += offscreen.json
 contains(QT_CONFIG, xcb):contains(QT_CONFIG, opengl):!contains(QT_CONFIG, opengles2) {
     SOURCES += qoffscreenintegration_x11.cpp
     HEADERS += qoffscreenintegration_x11.h
+    LIBS_PRIVATE += -lXrender -lX11 -lfreetype $$QT_LIBS_GLIB
     system(echo "Using X11 offscreen integration with GLX")
 } else {
     SOURCES += qoffscreenintegration_dummy.cpp
--- origsrc/qtbase-opensource-src-5.1.0/src/plugins/platforms/xcb/xcb-plugin.pro	2013-07-02 02:09:53.000000000 -0500
+++ src/qtbase-opensource-src-5.1.0/src/plugins/platforms/xcb/xcb-plugin.pro	2013-08-14 16:32:24.458988200 -0500
@@ -93,6 +93,7 @@ contains(QT_CONFIG, opengl) {
 
 DEFINES += $$QMAKE_DEFINES_XCB
 LIBS += $$QMAKE_LIBS_XCB
+LIBS_PRIVATE += -lXrender $$QMAKE_LIBS_FONTCONFIG $$QT_LIBS_GLIB
 QMAKE_CXXFLAGS += $$QMAKE_CFLAGS_XCB
 
 CONFIG += qpa/genericunixfontdatabase
