--- origsrc/qtbase-opensource-src-5.1.0/src/corelib/global/qglobal.cpp	2013-07-02 02:10:01.000000000 -0500
+++ src/qtbase-opensource-src-5.1.0/src/corelib/global/qglobal.cpp	2013-08-14 12:33:02.152511300 -0500
@@ -1706,7 +1706,7 @@ QSysInfo::MacVersion QSysInfo::macVersio
 }
 const QSysInfo::MacVersion QSysInfo::MacintoshVersion = QSysInfo::macVersion();
 
-#elif defined(Q_OS_WIN) || defined(Q_OS_CYGWIN) || defined(Q_OS_WINCE) || defined(Q_OS_WINRT)
+#elif defined(Q_OS_WIN) || defined(Q_OS_WINCE) || defined(Q_OS_WINRT)
 
 QT_BEGIN_INCLUDE_NAMESPACE
 #include "qt_windows.h"
--- origsrc/qtbase-opensource-src-5.1.0/src/corelib/global/qsysinfo.h	2013-07-02 02:10:01.000000000 -0500
+++ src/qtbase-opensource-src-5.1.0/src/corelib/global/qsysinfo.h	2013-08-14 12:33:24.228774000 -0500
@@ -80,7 +80,7 @@ public:
 #  endif
     };
 #endif
-#if defined(Q_OS_WIN) || defined(Q_OS_CYGWIN)
+#if defined(Q_OS_WIN)
     enum WinVersion {
         WV_32s      = 0x0001,
         WV_95       = 0x0002,
--- origsrc/qtbase-opensource-src-5.1.0/src/corelib/io/qfilesystemiterator_p.h	2013-07-02 02:10:02.000000000 -0500
+++ src/qtbase-opensource-src-5.1.0/src/corelib/io/qfilesystemiterator_p.h	2013-08-14 12:34:06.285179400 -0500
@@ -95,7 +95,7 @@ private:
 #else
     QT_DIR *dir;
     QT_DIRENT *dirEntry;
-#if defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(Q_OS_CYGWIN)
+#if defined(_POSIX_THREAD_SAFE_FUNCTIONS)
     // for readdir_r
     QScopedPointer<QT_DIRENT, QScopedPointerPodDeleter> mt_file;
 #if defined(Q_OS_QNX) && defined(__EXT_QNX__READDIR_R)
--- origsrc/qtbase-opensource-src-5.1.0/src/corelib/io/qfilesystemiterator_unix.cpp	2013-07-02 02:10:03.000000000 -0500
+++ src/qtbase-opensource-src-5.1.0/src/corelib/io/qfilesystemiterator_unix.cpp	2013-08-14 12:34:23.552167100 -0500
@@ -70,7 +70,7 @@ QFileSystemIterator::QFileSystemIterator
         if (!nativePath.endsWith('/'))
             nativePath.append('/');
 
-#if defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(Q_OS_CYGWIN)
+#if defined(_POSIX_THREAD_SAFE_FUNCTIONS)
         // ### Race condition; we should use fpathconf and dirfd().
         size_t maxPathName = ::pathconf(nativePath.constData(), _PC_NAME_MAX);
         if (maxPathName == size_t(-1))
@@ -109,7 +109,7 @@ bool QFileSystemIterator::advance(QFileS
     lastError = _readdir_r(dir, mt_file.data(), &dirEntry, direntSize);
     if (lastError)
         return false;
-#elif defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(Q_OS_CYGWIN)
+#elif defined(_POSIX_THREAD_SAFE_FUNCTIONS)
     lastError = QT_READDIR_R(dir, mt_file.data(), &dirEntry);
     if (lastError)
         return false;
--- origsrc/qtbase-opensource-src-5.1.0/src/widgets/kernel/qapplication_p.h	2013-07-02 02:09:50.000000000 -0500
+++ src/qtbase-opensource-src-5.1.0/src/widgets/kernel/qapplication_p.h	2013-08-14 12:34:48.478592800 -0500
@@ -92,7 +92,7 @@ extern Q_GUI_EXPORT bool qt_is_gui_used;
 extern QClipboard *qt_clipboard;
 #endif
 
-#if defined (Q_OS_WIN32) || defined (Q_OS_CYGWIN) || defined(Q_OS_WINCE)
+#if defined (Q_OS_WIN32) || defined(Q_OS_WINCE)
 extern QSysInfo::WinVersion qt_winver;
 # ifdef Q_OS_WINCE
   extern DWORD qt_cever;
