--- origsrc/qtbase-opensource-src-5.2.1/src/opengl/qgl.cpp	2014-02-01 14:37:30.000000000 -0600
+++ src/qtbase-opensource-src-5.2.1/src/opengl/qgl.cpp	2014-05-12 19:21:28.552129600 -0500
@@ -4772,6 +4772,8 @@ const QString qt_gl_library_name()
     if (qt_gl_lib_name()->isNull()) {
 # if defined(QT_OPENGL_ES_2)
         return QLatin1String("GLESv2");
+# elif defined(Q_OS_CYGWIN)
+        return QLatin1String("cygGL-1.dll");
 # else
         return QLatin1String("GL");
 # endif
--- origsrc/qtbase-opensource-src-5.2.1/src/plugins/platforms/xcb/qglxintegration.cpp	2014-02-01 14:37:36.000000000 -0600
+++ src/qtbase-opensource-src-5.2.1/src/plugins/platforms/xcb/qglxintegration.cpp	2014-05-12 19:24:56.300510300 -0500
@@ -382,7 +382,11 @@ void (*QGLXContext::getProcAddress(const
             {
                 extern const QString qt_gl_library_name();
 //                QLibrary lib(qt_gl_library_name());
+#if defined(Q_OS_CYGWIN)
+                QLibrary lib(QLatin1String("cygGL-1.dll"));
+#else
                 QLibrary lib(QLatin1String("GL"));
+#endif
                 glXGetProcAddressARB = (qt_glXGetProcAddressARB) lib.resolve("glXGetProcAddressARB");
             }
         }
