--- origsrc/qtbase-opensource-src-5.3.2/configure	2014-09-11 05:47:55.000000000 -0500
+++ src/qtbase-opensource-src-5.3.2/configure	2014-10-24 12:06:50.843762700 -0500
@@ -3773,7 +3773,7 @@ if true; then ###[ '!' -f "$outpath/bin/
             echo "-include \$(notdir \$(DEPEND_SRC:%.cpp=%.d))" >> "$mkfile"
         else
             cat "$in_mkfile" >> "$mkfile"
-            if "$WHICH" makedepend >/dev/null 2>&1 && grep 'depend:' "$mkfile" >/dev/null 2>&1; then
+            if false && "$WHICH" makedepend >/dev/null 2>&1 && grep 'depend:' "$mkfile" >/dev/null 2>&1; then
                 (cd "$outpath/qmake" && "$MAKE" -f "$mkfile" depend) >/dev/null 2>&1
                 sed 's,^.*/\([^/]*.o\):,\1:,g' "$mkfile" >"$mkfile.tmp"
                 sed "s,$outpath,$adjoutpath,g" "$mkfile.tmp" >"$mkfile"
