--- origsrc/qtbase-opensource-src-5.3.2/mkspecs/features/qt_tool.prf	2014-09-11 05:48:07.000000000 -0500
+++ src/qtbase-opensource-src-5.3.2/mkspecs/features/qt_tool.prf	2014-10-24 12:08:03.923042600 -0500
@@ -63,3 +63,5 @@ CONFIG += console
         cache(QT_TOOL.$${MODULE}.$$var, transient)
 
 }
+
+LIBS_PRIVATE += -lz
--- origsrc/qtbase-opensource-src-5.3.2/src/plugins/platforms/minimal/minimal.pro	2014-09-11 05:48:06.000000000 -0500
+++ src/qtbase-opensource-src-5.3.2/src/plugins/platforms/minimal/minimal.pro	2014-10-24 12:08:03.928043300 -0500
@@ -6,6 +6,7 @@ PLUGIN_CLASS_NAME = QMinimalIntegrationP
 load(qt_plugin)
 
 QT += core-private gui-private platformsupport-private
+LIBS_PRIVATE += $$QT_LIBS_GLIB
 
 SOURCES =   main.cpp \
             qminimalintegration.cpp \
--- origsrc/qtbase-opensource-src-5.3.2/src/plugins/platforms/offscreen/offscreen.pro	2014-09-11 05:48:06.000000000 -0500
+++ src/qtbase-opensource-src-5.3.2/src/plugins/platforms/offscreen/offscreen.pro	2014-10-24 12:08:03.933043900 -0500
@@ -21,6 +21,7 @@ OTHER_FILES += offscreen.json
 contains(QT_CONFIG, xlib):contains(QT_CONFIG, opengl):!contains(QT_CONFIG, opengles2) {
     SOURCES += qoffscreenintegration_x11.cpp
     HEADERS += qoffscreenintegration_x11.h
+    LIBS_PRIVATE += -lXrender -lX11 -lfreetype $$QT_LIBS_GLIB
     system(echo "Using X11 offscreen integration with GLX")
 } else {
     SOURCES += qoffscreenintegration_dummy.cpp
--- origsrc/qtbase-opensource-src-5.3.2/src/plugins/platforms/xcb/xcb-plugin.pro	2014-09-11 05:48:06.000000000 -0500
+++ src/qtbase-opensource-src-5.3.2/src/plugins/platforms/xcb/xcb-plugin.pro	2014-10-24 12:08:03.937544500 -0500
@@ -104,6 +104,7 @@ contains(QT_CONFIG, opengl) {
 
 DEFINES += $$QMAKE_DEFINES_XCB
 LIBS += $$QMAKE_LIBS_XCB
+LIBS_PRIVATE += -lXrender $$QMAKE_LIBS_FONTCONFIG $$QT_LIBS_GLIB
 QMAKE_CXXFLAGS += $$QMAKE_CFLAGS_XCB
 
 CONFIG += qpa/genericunixfontdatabase
