--- origsrc/qtbase-opensource-src-5.5.1/configure	2015-10-12 23:35:07.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/configure	2015-10-18 17:20:12.537075000 -0500
@@ -4053,7 +4053,7 @@ if true; then ###[ '!' -f "$outpath/bin/
             echo "-include \$(notdir \$(DEPEND_SRC:%.cpp=%.d))" >> "$mkfile"
         else
             cat "$in_mkfile" >> "$mkfile"
-            if "$WHICH" makedepend >/dev/null 2>&1 && grep 'depend:' "$mkfile" >/dev/null 2>&1; then
+            if false && "$WHICH" makedepend >/dev/null 2>&1 && grep 'depend:' "$mkfile" >/dev/null 2>&1; then
                 (cd "$outpath/qmake" && "$MAKE" -f "$mkfile" depend) >/dev/null 2>&1
                 sed 's,^.*/\([^/]*.o\):,\1:,g' "$mkfile" >"$mkfile.tmp"
                 sed "s,$outpath,$adjoutpath,g" "$mkfile.tmp" >"$mkfile"
