--- origsrc/qtbase-opensource-src-5.5.1/mkspecs/features/qt_tool.prf	2015-10-12 23:35:32.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/mkspecs/features/qt_tool.prf	2015-10-18 17:21:18.984875600 -0500
@@ -63,3 +63,5 @@ CONFIG += console
         cache(QT_TOOL.$${MODULE}.$$var, transient)
 
 }
+
+LIBS_PRIVATE += -lz
--- origsrc/qtbase-opensource-src-5.5.1/src/plugins/platforms/minimal/minimal.pro	2015-10-12 23:35:27.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/plugins/platforms/minimal/minimal.pro	2015-10-18 17:21:18.988875800 -0500
@@ -6,6 +6,7 @@ PLUGIN_CLASS_NAME = QMinimalIntegrationP
 load(qt_plugin)
 
 QT += core-private gui-private platformsupport-private
+LIBS_PRIVATE += $$QT_LIBS_GLIB
 
 SOURCES =   main.cpp \
             qminimalintegration.cpp \
--- origsrc/qtbase-opensource-src-5.5.1/src/plugins/platforms/offscreen/offscreen.pro	2015-10-12 23:35:28.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/plugins/platforms/offscreen/offscreen.pro	2015-10-18 17:21:19.001876600 -0500
@@ -21,6 +21,7 @@ OTHER_FILES += offscreen.json
 contains(QT_CONFIG, xlib):contains(QT_CONFIG, opengl):!contains(QT_CONFIG, opengles2) {
     SOURCES += qoffscreenintegration_x11.cpp
     HEADERS += qoffscreenintegration_x11.h
+    LIBS_PRIVATE += -lXrender -lX11 -lfreetype $$QT_LIBS_GLIB
     system(echo "Using X11 offscreen integration with GLX")
 } else {
     SOURCES += qoffscreenintegration_dummy.cpp
--- origsrc/qtbase-opensource-src-5.5.1/src/plugins/platforms/xcb/gl_integrations/xcb_glx/xcb_glx.pro	2015-10-12 23:35:27.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/plugins/platforms/xcb/gl_integrations/xcb_glx/xcb_glx.pro	2015-10-18 18:37:48.172166600 -0500
@@ -11,6 +11,7 @@ include(../gl_integrations_plugin_base.p
 DEFINES += XCB_USE_GLX XCB_USE_XLIB
 
 LIBS += -lxcb
+LIBS_PRIVATE += -lXrender
 
 contains(QT_CONFIG, xcb-glx) {
     DEFINES += XCB_HAS_XCB_GLX
--- origsrc/qtbase-opensource-src-5.5.1/src/plugins/platforms/xcb/xcb_qpa_lib.pro	2015-10-12 23:35:27.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/plugins/platforms/xcb/xcb_qpa_lib.pro	2015-10-18 18:31:44.087933700 -0500
@@ -48,6 +48,7 @@ HEADERS = \
         qxcbsystemtraytracker.h
 
 LIBS += $$QMAKE_LIBS_DYNLOAD
+LIBS_PRIVATE += $$QMAKE_LIBS_FONTCONFIG $$QT_LIBS_GLIB
 
 DEFINES += QT_BUILD_XCB_PLUGIN
 # needed by Xcursor ...
--- origsrc/qtbase-opensource-src-5.5.1/src/plugins/platformthemes/gtk2/gtk2.pro	2015-10-12 23:35:28.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/plugins/platformthemes/gtk2/gtk2.pro	2015-10-18 18:39:30.471657000 -0500
@@ -5,7 +5,7 @@ PLUGIN_EXTENDS = -
 PLUGIN_CLASS_NAME = QGtk2ThemePlugin
 load(qt_plugin)
 
-QT += core-private gui-private platformsupport-private
+QT += core-private gui-private dbus-private platformsupport-private
 
 CONFIG += X11
 QMAKE_CXXFLAGS += $$QT_CFLAGS_QGTK2
