--- origsrc/qtbase-opensource-src-5.5.1/src/corelib/global/qcompilerdetection.h	2015-10-12 23:35:31.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/corelib/global/qcompilerdetection.h	2015-10-18 17:21:42.744234500 -0500
@@ -201,7 +201,7 @@
 #    endif
 #  endif
 
-#  ifdef Q_OS_WIN
+#  if defined(Q_OS_WIN) || defined(Q_OS_CYGWIN)
 #    define Q_DECL_EXPORT     __declspec(dllexport)
 #    define Q_DECL_IMPORT     __declspec(dllimport)
 #  elif defined(QT_VISIBILITY_AVAILABLE)
--- origsrc/qtbase-opensource-src-5.5.1/src/gui/text/qfontengine_ft_p.h	2015-10-12 23:35:30.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/gui/text/qfontengine_ft_p.h	2015-10-18 17:21:42.751234900 -0500
@@ -121,7 +121,7 @@ private:
 
 // If this is exported this breaks compilation of the windows
 // plugin as qfontengine_ft_p.h/.cpp are also compiled there
-#ifndef Q_OS_WIN
+#if !defined(Q_OS_WIN) && !defined(Q_OS_CYGWIN)
 class Q_GUI_EXPORT QFontEngineFT : public QFontEngine
 #else
 class QFontEngineFT : public QFontEngine
--- origsrc/qtbase-opensource-src-5.5.1/src/plugins/platforms/xcb/qxcbscreen.h	2015-10-12 23:35:27.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/plugins/platforms/xcb/qxcbscreen.h	2015-10-18 18:35:34.842235800 -0500
@@ -184,7 +184,7 @@ private:
 };
 
 #ifndef QT_NO_DEBUG_STREAM
-Q_GUI_EXPORT QDebug operator<<(QDebug, const QXcbScreen *);
+Q_XCB_EXPORT QDebug operator<<(QDebug, const QXcbScreen *);
 #endif
 
 QT_END_NAMESPACE
