These must use a different codepath on glibc

--- origsrc/qtbase-opensource-src-5.5.1/src/corelib/io/forkfd_qt.cpp	2015-10-12 23:35:31.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/corelib/io/forkfd_qt.cpp	2016-09-01 21:00:22.397262400 -0500
@@ -38,6 +38,7 @@
 #if !defined(_XOPEN_SOURCE) && !defined(__QNXNTO__) && !defined(ANDROID)
 #  define _XOPEN_SOURCE 700
 #endif
+#define _GNU_SOURCE 1
 
 #include <QtCore/qatomic.h>
 #include "qprocess_p.h"
--- origsrc/qtbase-opensource-src-5.5.1/src/corelib/tools/qelapsedtimer_unix.cpp	2015-10-12 23:35:31.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/corelib/tools/qelapsedtimer_unix.cpp	2016-09-01 20:59:32.790222500 -0500
@@ -33,6 +33,7 @@
 
 // ask for the latest POSIX, just in case
 #define _POSIX_C_SOURCE 200809L
+#define _GNU_SOURCE 1
 
 #include "qelapsedtimer.h"
 #if defined(Q_OS_VXWORKS)
