--- origsrc/qtbase-opensource-src-5.5.1/src/corelib/global/qglobal.cpp	2015-10-12 23:35:31.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/corelib/global/qglobal.cpp	2015-10-18 17:20:00.473385000 -0500
@@ -1878,7 +1878,7 @@ QSysInfo::MacVersion QSysInfo::macVersio
 }
 const QSysInfo::MacVersion QSysInfo::MacintoshVersion = QSysInfo::macVersion();
 
-#elif defined(Q_OS_WIN) || defined(Q_OS_CYGWIN) || defined(Q_OS_WINCE) || defined(Q_OS_WINRT)
+#elif defined(Q_OS_WIN) || defined(Q_OS_WINCE) || defined(Q_OS_WINRT)
 
 QT_BEGIN_INCLUDE_NAMESPACE
 #include "qt_windows.h"
--- origsrc/qtbase-opensource-src-5.5.1/src/corelib/global/qsysinfo.h	2015-10-12 23:35:31.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/corelib/global/qsysinfo.h	2015-10-18 17:19:53.056960800 -0500
@@ -110,7 +110,7 @@ public:
         WV_CE_6     = 0x0400,
         WV_CE_based = 0x0f00
     };
-#if defined(Q_OS_WIN) || defined(Q_OS_CYGWIN)
+#if defined(Q_OS_WIN)
     static const WinVersion WindowsVersion;
     static WinVersion windowsVersion();
 #else
--- origsrc/qtbase-opensource-src-5.5.1/src/corelib/io/qfilesystemiterator_p.h	2015-10-12 23:35:31.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/corelib/io/qfilesystemiterator_p.h	2015-10-18 17:20:00.488385800 -0500
@@ -87,7 +87,7 @@ private:
 #else
     QT_DIR *dir;
     QT_DIRENT *dirEntry;
-#if defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(Q_OS_CYGWIN) || defined(QT_EXT_QNX_READDIR_R)
+#if defined(_POSIX_THREAD_SAFE_FUNCTIONS) || defined(QT_EXT_QNX_READDIR_R)
     // for readdir_r
     QScopedPointer<QT_DIRENT, QScopedPointerPodDeleter> mt_file;
 #if defined(QT_EXT_QNX_READDIR_R)
--- origsrc/qtbase-opensource-src-5.5.1/src/corelib/io/qfilesystemiterator_unix.cpp	2015-10-12 23:35:31.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/corelib/io/qfilesystemiterator_unix.cpp	2015-10-18 17:20:00.494386200 -0500
@@ -62,7 +62,7 @@ QFileSystemIterator::QFileSystemIterator
         if (!nativePath.endsWith('/'))
             nativePath.append('/');
 
-#if defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(Q_OS_CYGWIN) || defined(QT_EXT_QNX_READDIR_R)
+#if defined(_POSIX_THREAD_SAFE_FUNCTIONS) || defined(QT_EXT_QNX_READDIR_R)
         // ### Race condition; we should use fpathconf and dirfd().
         size_t maxPathName = ::pathconf(nativePath.constData(), _PC_NAME_MAX);
         if (maxPathName == size_t(-1))
@@ -102,7 +102,7 @@ bool QFileSystemIterator::advance(QFileS
     lastError = QT_EXT_QNX_READDIR_R(dir, mt_file.data(), &dirEntry, direntSize);
     if (lastError)
         return false;
-#elif defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(Q_OS_CYGWIN)
+#elif defined(_POSIX_THREAD_SAFE_FUNCTIONS)
     lastError = QT_READDIR_R(dir, mt_file.data(), &dirEntry);
     if (lastError)
         return false;
--- origsrc/qtbase-opensource-src-5.5.1/src/widgets/kernel/qapplication_p.h	2015-10-12 23:35:22.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/widgets/kernel/qapplication_p.h	2015-10-18 17:20:00.500386500 -0500
@@ -84,7 +84,7 @@ extern Q_GUI_EXPORT bool qt_is_gui_used;
 extern QClipboard *qt_clipboard;
 #endif
 
-#if defined (Q_OS_WIN32) || defined (Q_OS_CYGWIN) || defined(Q_OS_WINCE)
+#if defined (Q_OS_WIN32) || defined(Q_OS_WINCE)
 extern QSysInfo::WinVersion qt_winver;
 # ifdef Q_OS_WINCE
   extern DWORD qt_cever;
