--- origsrc/qtbase-opensource-src-5.6.2/src/3rdparty/forkfd/forkfd.c	2016-09-16 00:49:42.000000000 -0500
+++ src/qtbase-opensource-src-5.6.2/src/3rdparty/forkfd/forkfd.c	2016-10-27 23:04:12.066127500 -0500
@@ -68,6 +68,11 @@
 #  undef HAVE_WAITID
 #endif
 
+#if defined(__CYGWIN__)
+#  undef HAVE_WAITID
+#  define HAVE_PIPE2	1
+#endif
+
 #if (defined(__FreeBSD__) && defined(__FreeBSD_version) && __FreeBSD_version >= 1000032) || \
     (defined(__OpenBSD__) && OpenBSD >= 201505) || \
     (defined(__NetBSD__) && __NetBSD_Version__ >= 600000000)
