--- origsrc/qtbase-opensource-src-5.6.2/mkspecs/features/qt_app.prf	2016-09-16 00:49:42.000000000 -0500
+++ src/qtbase-opensource-src-5.6.2/mkspecs/features/qt_app.prf	2016-10-28 14:59:34.413919100 -0500
@@ -25,6 +25,7 @@ host_build:force_bootstrap {
     contains(QT, core(-private)?|xml) {
         QT -= core core-private xml
         QT += bootstrap-private
+        LIBS_PRIVATE += -lz
     }
     target.path = $$[QT_HOST_BINS]
 } else {
--- origsrc/qtbase-opensource-src-5.6.2/src/platformsupport/platformsupport.pro	2016-09-16 00:49:42.000000000 -0500
+++ src/qtbase-opensource-src-5.6.2/src/platformsupport/platformsupport.pro	2016-10-27 23:26:31.497479800 -0500
@@ -2,7 +2,7 @@ TARGET     = QtPlatformSupport
 QT         = core-private gui-private
 
 CONFIG += static internal_module
-mac:LIBS_PRIVATE += -lz
+LIBS_PRIVATE += -lz
 
 DEFINES += QT_NO_CAST_FROM_ASCII
 PRECOMPILED_HEADER = ../corelib/global/qt_pch.h
--- origsrc/qtbase-opensource-src-5.6.2/src/plugins/platforms/minimal/minimal.pro	2016-09-16 00:49:42.000000000 -0500
+++ src/qtbase-opensource-src-5.6.2/src/plugins/platforms/minimal/minimal.pro	2016-10-28 16:37:00.761385200 -0500
@@ -1,6 +1,7 @@
 TARGET = qminimal
 
 QT += core-private gui-private platformsupport-private
+contains(QT_CONFIG, glib):LIBS_PRIVATE += $$QT_LIBS_GLIB
 
 SOURCES =   main.cpp \
             qminimalintegration.cpp \
--- origsrc/qtbase-opensource-src-5.6.2/src/plugins/platforms/offscreen/offscreen.pro	2016-09-16 00:49:42.000000000 -0500
+++ src/qtbase-opensource-src-5.6.2/src/plugins/platforms/offscreen/offscreen.pro	2016-10-28 16:37:52.736539300 -0500
@@ -16,6 +16,8 @@ OTHER_FILES += offscreen.json
 contains(QT_CONFIG, xlib):contains(QT_CONFIG, opengl):!contains(QT_CONFIG, opengles2) {
     SOURCES += qoffscreenintegration_x11.cpp
     HEADERS += qoffscreenintegration_x11.h
+    LIBS_PRIVATE += -lXrender -lX11 -lfreetype
+    contains(QT_CONFIG, glib):LIBS_PRIVATE += $$QT_LIBS_GLIB
     system(echo "Using X11 offscreen integration with GLX")
 } else {
     SOURCES += qoffscreenintegration_dummy.cpp
--- origsrc/qtbase-opensource-src-5.6.2/src/plugins/platforms/xcb/gl_integrations/xcb_glx/xcb_glx.pro	2016-09-16 00:49:42.000000000 -0500
+++ src/qtbase-opensource-src-5.6.2/src/plugins/platforms/xcb/gl_integrations/xcb_glx/xcb_glx.pro	2016-10-28 16:39:20.150412800 -0500
@@ -6,6 +6,7 @@ include(../gl_integrations_plugin_base.p
 DEFINES += XCB_USE_GLX XCB_USE_XLIB
 
 LIBS += -lxcb
+LIBS_PRIVATE += -lXrender -lX11
 
 contains(QT_CONFIG, xcb-glx) {
     DEFINES += XCB_HAS_XCB_GLX
--- origsrc/qtbase-opensource-src-5.6.2/src/plugins/platforms/xcb/xcb_qpa_lib.pro	2016-09-16 00:49:42.000000000 -0500
+++ src/qtbase-opensource-src-5.6.2/src/plugins/platforms/xcb/xcb_qpa_lib.pro	2016-10-28 16:42:25.055794100 -0500
@@ -105,4 +105,6 @@ contains(QT_CONFIG, xkbcommon-qt) {
     QMAKE_CXXFLAGS += $$QMAKE_CFLAGS_XKBCOMMON
 }
 
+contains(QT_CONFIG, glib):LIBS_PRIVATE += $$QT_LIBS_GLIB
+
 load(qt_module)
--- origsrc/qtbase-opensource-src-5.6.2/src/plugins/platformthemes/gtk2/gtk2.pro	2016-09-16 00:49:42.000000000 -0500
+++ src/qtbase-opensource-src-5.6.2/src/plugins/platformthemes/gtk2/gtk2.pro	2016-10-28 16:41:08.325363000 -0500
@@ -1,6 +1,6 @@
 TARGET = qgtk2
 
-QT += core-private gui-private platformsupport-private
+QT += core-private gui-private dbus-private platformsupport-private
 
 CONFIG += X11
 QMAKE_CXXFLAGS += $$QT_CFLAGS_QGTK2
