--- origsrc/qtbase-opensource-src-5.9.2/src/platformsupport/services/genericunix/qgenericunixservices.cpp	2017-10-02 03:43:38.000000000 -0500
+++ src/qtbase-opensource-src-5.9.2/src/platformsupport/services/genericunix/qgenericunixservices.cpp	2017-10-22 00:07:55.962699100 -0500
@@ -61,13 +61,16 @@ static inline QByteArray detectDesktopEn
 {
     const QByteArray xdgCurrentDesktop = qgetenv("XDG_CURRENT_DESKTOP");
     if (!xdgCurrentDesktop.isEmpty())
-        return xdgCurrentDesktop.toUpper(); // KDE, GNOME, UNITY, LXDE, MATE, XFCE...
+        return xdgCurrentDesktop.contains("GNOME") ? "GNOME" // GNOME, GNOME Classic, GNOME Flashback
+                                                   : xdgCurrentDesktop.toUpper(); // KDE, UNITY, LXDE, MATE, XFCE...
 
     // Classic fallbacks
     if (!qEnvironmentVariableIsEmpty("KDE_FULL_SESSION"))
         return QByteArrayLiteral("KDE");
     if (!qEnvironmentVariableIsEmpty("GNOME_DESKTOP_SESSION_ID"))
         return QByteArrayLiteral("GNOME");
+    if (!qEnvironmentVariableIsEmpty("MATE_DESKTOP_SESSION_ID"))
+        return QByteArrayLiteral("MATE");
 
     // Fallback to checking $DESKTOP_SESSION (unreliable)
     QByteArray desktopSession = qgetenv("DESKTOP_SESSION");
@@ -127,8 +130,13 @@ static inline bool detectWebBrowser(cons
             browser->append(QLatin1String(" exec"));
             return true;
         }
-    } else if (desktop == QByteArray("GNOME")) {
-        if (checkExecutable(QStringLiteral("gnome-open"), browser))
+    } else if (desktop == QByteArray("GNOME") ||
+               desktop == QByteArray("LXDE") ||
+               desktop == QByteArray("LXQT") ||
+               desktop == QByteArray("MATE") ||
+               desktop == QByteArray("X-CINNAMON") ||
+               desktop == QByteArray("XFCE")) {
+        if (checkExecutable(QStringLiteral("gvfs-open"), browser))
             return true;
     }
 
--- origsrc/qtbase-opensource-src-5.9.2/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp	2017-10-02 03:43:38.000000000 -0500
+++ src/qtbase-opensource-src-5.9.2/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp	2017-10-22 00:07:55.962699100 -0500
@@ -212,6 +212,8 @@ QPlatformSystemTrayIcon *QGenericUnixThe
 QVariant QGenericUnixTheme::themeHint(ThemeHint hint) const
 {
     switch (hint) {
+    case QPlatformTheme::SystemIconThemeName:
+        return QVariant(QString(QStringLiteral("oxygen")));
     case QPlatformTheme::SystemIconFallbackThemeName:
         return QVariant(QString(QStringLiteral("hicolor")));
     case QPlatformTheme::IconThemeSearchPaths:
@@ -848,6 +850,7 @@ QStringList QGenericUnixTheme::themeName
         QList<QByteArray> gtkBasedEnvironments;
         gtkBasedEnvironments << "GNOME"
                              << "X-CINNAMON"
+                             << "X-CYGWIN"
                              << "UNITY"
                              << "MATE"
                              << "XFCE"
@@ -859,6 +862,8 @@ QStringList QGenericUnixTheme::themeName
                 result.push_back(QLatin1String(QKdeTheme::name));
 #endif
             } else if (gtkBasedEnvironments.contains(desktopName)) {
+                if (desktopEnvironment == "GNOME")
+                    result.push_back(QStringLiteral("qgnomeplatform"));
                 // prefer the GTK3 theme implementation with native dialogs etc.
                 result.push_back(QStringLiteral("gtk3"));
                 // fallback to the generic Gnome theme if loading the GTK3 theme fails
